test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+            var params = datasets.partition2(users);
+
+            it('should match types of passing elements', function () {
+                var element = typeof params[1];
+                expect(element).to.eql('string');
+                expect(element).to.have.length.above(1);
+                expect(params.length).to.eql(2);
+            });
+            it('should split persons only to truthy table when predicate is true for all', function () {
+                expect(_.partition.apply(_, datasets.partition2(users))).to.eql([[
+                    {name: 'Andrinio', age: 36, active: false},
+                    {name: 'Harry', age: 40, active: true},
+                    {name: 'Kodi', age: 1, active: false},
+                    {name: 'Francheska', age: 61, active: true}],
+                    []]);
+            });
+        });
+        describe('partition3 function', function () {
+            var params = datasets.partition3(users);
+            it('should match types of passing elements', function () {
+                expect(typeof params[1]).to.be.oneOf(['number', 'string', 'object', 'boolean']);
+            });
+            it('should split persons only to falsy table when predicate is false for all', function () {
+                expect(_.partition.apply(_, datasets.partition3(users))).to.eql([[],
+                    [
+                        {name: 'Andrinio', age: 36, active: false},
+                        {name: 'Harry', age: 40, active: true},
+                        {name: 'Kodi', age: 1, active: false},
+                        {name: 'Francheska', age: 61, active: true}
+                    ]
+                ]);
+            });
+        });
+    });
+
+    describe('reduce', function () {
+        var params = datasets.reduce();
+        var array = params[0];
+        var spy = sinon.spy(params[1]);
+
+        it('should match types of passing elements', function () {
+            var element1 = params[0] instanceof Array;
+            var element2 = params[1] instanceof Function;
+            var element3 = typeof params[2];
+
+            expect(element1).to.eql(true);
+            expect(element2).to.eql(true);
+            expect(element3).to.eql('number')
+            expect(params.length).to.eql(3);
+        });
+        it('should pass mock test', function () {
+            var testArray = [10, 11, 12, 15];
+            var accumulator = 0;
+            var checkFunc = _.reduce.apply(_, [testArray, spy, accumulator]);
+            expect(spy).to.have.been.callCount(4);
+            expect(checkFunc).to.eql(590);
+        });
+        it('should count sum of squared numbers', function () {
+            array.forEach(function (e) {
+                expect(typeof e).to.eql('number');
+            });
+            expect(array.length).to.eql(5);
+            expect(array).to.include(1);
+            expect(array).to.include(5);
+            expect(_.reduce.apply(_, datasets.reduce())).to.eql(55);
```

Comment:
I can easily hack the answer by simply returning hardcoded values based on my input array. To test it properly you would need to use some random numbers(or another creative idea). Overally exercise doesn't show what accumulator is for and it is quite poor.